dev/breeze/src/airflow_breeze/utils/docker_command_utils.py

```diff
@@ -53,28 +67,175 @@
+        else:
+            good_version = compare_version(docker_version, MIN_DOCKER_VERSION)
+            if good_version:
+                console.print(f'Good version of Docker: {docker_version}.')
+            else:
+                console.print(
+                    f'Your version of docker is too old:{docker_version}. Please upgrade to \
+                    at least ${MIN_DOCKER_VERSION}'
+                )
+
+
+def check_docker_compose_version(verbose: bool):
+    version_pattern = re.compile(r'(\d+)\.(\d+)\.(\d+)')
+    docker_compose_version_command = ["docker-compose", "--version"]
+    docker_compose_version_output = run_command(
+        docker_compose_version_command,
+        verbose=verbose,
+        suppress_console_print=True,
+        capture_output=True,
+        text=True,
+    )
+    if docker_compose_version_output.returncode == 0:
+        docker_compose_version = docker_compose_version_output.stdout
+        version_extracted = version_pattern.search(docker_compose_version)
+        if version_extracted is not None:
+            version = '.'.join(version_extracted.groups())
+            good_version = compare_version(version, MIN_DOCKER_COMPOSE_VERSION)
+            if good_version:
+                console.print(f'Good version of docker-compose: {version}')
+            else:
+                console.print(
+                    f'You have too old version of docker-compose: {version}! \
+                At least 1.29 is needed! Please upgrade!'
+                )
+                console.print(
+                    'See https://docs.docker.com/compose/install/ for instructions. \
+                Make sure docker-compose you install is first on the PATH variable of yours.'
+                )
+    else:
+        console.print(
+            'Unknown docker-compose version. At least 1.29 is needed! \
+        If Breeze fails upgrade to latest available docker-compose version'
+        )
+
+
+def prepare_docker_compose_files(verbose: bool, backend: str, integrations: Tuple[str]):
+    # fetch backend and integration from cache
+    compose_ci_file = []
+    main_ci_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/base.yml"
+    backend_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/backend-{backend}.yml"
+    backend_port_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/backend-{backend}-port.yml"
+    local_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/local.yml"
+    local_all_sources_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/local-all-sources.yml"
+    files_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/files.yml"
+    remove_sources_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/remove-sources.yml"
+    forward_credentials_docker_compose_file = f"{str(SCRIPTS_CI_DIR)}/docker-compose/forward-credentials.yml"
+    # mssql based check have to be added
+    if backend == 'mssql':
+        docker_filesystem_command = [
```

Comment:
@potiuk I face error when executing this command via subprocess. While it works via terminal, it throws error when executed via subprocess. Can you help me to solve this issue?

Reply:
I need to see it - ping me on slack or better schedule 1-1 session, I think this is about the right time to have a session with entering the shell as I can help to resolve it quicker by chatting.